src/Nethermind/Nethermind.Evm/Precompiles/Snarks/Bn254PairingPrecompile.cs

```diff
@@ -37,10 +38,13 @@ public class Bn254PairingPrecompile : IPrecompile<Bn254PairingPrecompile>
             return IPrecompile.Failure;
         }
 
+        byte[] inputDataArray = ArrayPool<byte>.Shared.Rent(inputData.Length);
```

Comment:
This isn't stack-allocated cause it's `calldata` and can cause an overflow, right?

Reply:
User input driven size; so smartcontract could crash the node if was just freeform

(could do double paring fixed stackalloc + slice below certain size and rent above)